keras_core/ops/numpy_test.py

```diff
@@ -285,6 +285,79 @@ def test_shape_equal_ignore_axes(self):
         self.assertTrue(knp.shape_equal(x, y, axis=(1, 2)))
         self.assertFalse(knp.shape_equal(x, y, axis=1))
 
+    def test_shape_equal_only_none(self):
+        x = KerasTensor([None, None]).shape
+        y = KerasTensor([5, 6]).shape
+        self.assertTrue(knp.shape_equal(x, y, allow_none=True))
+
+    def test_shape_equal_axis_as_list(self):
+        x = KerasTensor([3, 4, 5]).shape
+        y = KerasTensor([3, 6, 5]).shape
+        self.assertTrue(knp.shape_equal(x, y, axis=[1]))
+
+    def test_shape_non_equal_with_negative_axis(self):
+        x = KerasTensor([3, 4, 5]).shape
+        y = KerasTensor([3, 4, 6]).shape
+        self.assertFalse(knp.shape_equal(x, y, axis=-2))
+
+    def test_shape_equal_with_negative_axis(self):
+        x = KerasTensor([3, 4, 5]).shape
+        y = KerasTensor([3, 4, 5]).shape
+        self.assertTrue(knp.shape_equal(x, y, axis=-1))
+
+    def test_shape_equal_zeros(self):
+        x = KerasTensor([0, 4]).shape
+        y = KerasTensor([0, 4]).shape
+        self.assertTrue(knp.shape_equal(x, y))
+        y = KerasTensor([0, 5]).shape
+        self.assertFalse(knp.shape_equal(x, y))
+
+    def test_broadcast_shapes_conversion_to_list(self):
+        shape1 = KerasTensor([1, 2]).shape
+        shape2 = KerasTensor([3, 1]).shape
+        expected_output = [3, 2]
+        self.assertEqual(knp.broadcast_shapes(shape1, shape2), expected_output)
+
+    def test_broadcast_shapes_shape1_longer_than_shape2(self):
+        shape1 = KerasTensor([5, 3, 2]).shape
+        shape2 = KerasTensor([1, 3]).shape
+        with self.assertRaises(ValueError):
```

Comment:
As a general best practice, always use `assertRaisesRegex` and match part of the error message. This ensures the correct message is being raised (and not an unrelated exception with the same type).